### DISCUSSION TITLE
Query Data from Folder which Contain Many CSV Files

### DISCUSSION THREAD

**Discussion by hkpeaks** [ClickHouse/ClickHouse#50627] [Q&A]
I would like to include ClickHouse with my coming benchmark exercise
https://github.com/duckdblabs/db-benchmark/issues/13#issuecomment-1578079219

Since I have no experience on this software, I would like to know how to write the script based on following requirements:-

Select{D:/Benchmark/Input/3000Files/*.csv | Ledger(L30..L70)}
GroupBy{Ledger, Account, DC,Currency => Sum(Base_Amount) ~ SummaryResult.csv}

Where L30..L70 mean Ledger >= L30 and Ledger <= L70

Bing Chat reply:

copy (SELECT Ledger, Account, DC, Currency, SUM(Base_Amount) as Total_Base_Amount
FROM read_csv_auto('input/3000Files/*.csv')
WHERE Ledger>='L30' AND Ledger <='L70'
GROUP BY Ledger, Account, DC, Currency)
to 'output/FilterGroupByCSV.csv' (format csv, header true);

**Comment by nickitat** [comment]
you need smth like

```
SELECT Ledger, Account, DC, Currency, SUM(Base_Amount) as Total_Base_Amount
FROM file('input/3000Files/*.csv')
WHERE Ledger>='L30' AND Ledger <='L70'
GROUP BY Ledger, Account, DC, Currency
into outfile 'output/FilterGroupByCSV.csv'
format CSVWithNames
```
